includes/base/element-base.php

```diff
@@ -720,6 +721,637 @@ protected function add_render_attributes() {
 		do_action( 'elementor/element/after_add_attributes', $this );
 	}
 
+	/**
+	 * Register the Transform controls in the advanced tab of the element.
+	 *
+	 * Previously registered under the Widget_Common class, but registered a more fundamental level now to enable access from other widgets.
+	 *
+	 * @since 3.9.0
+	 * @access protected
+	 * @return void
+	 */
+	protected function register_transform_section( $element_selector = '' ) {
+		$default_unit_values_deg = [];
+		$default_unit_values_ms = [];
+
+		// Set the default unit sizes for all active breakpoints.
+		foreach ( Breakpoints_Manager::get_default_config() as $breakpoint_name => $breakpoint_config ) {
+			$default_unit_values_deg[ $breakpoint_name ] = [
+				'default' => [
+					'unit' => 'deg',
+				],
+			];
+
+			$default_unit_values_ms[ $breakpoint_name ] = [
+				'default' => [
+					'unit' => 'ms',
+				],
+			];
+		}
+
+		$this->start_controls_section(
+			'_section_transform',
+			[
+				'label' => esc_html__( 'Transform', 'elementor' ),
+				'tab' => Controls_Manager::TAB_ADVANCED,
+			]
+		);
+
+		$this->start_controls_tabs( '_tabs_positioning' );
+
+		$transform_prefix_class = 'e-';
+		$transform_return_value = 'transform';
+		$transform_selector_class = ' > .elementor-widget-container';
+		$transform_css_modifier = '';
+
+		if ( 'container' === $element_selector ) {
+			$transform_selector_class = '.e-con';
+			$transform_css_modifier = 'con-';
+		}
+
+		foreach ( [ '', '_hover' ] as $tab ) {
+			$state = '_hover' === $tab ? ':hover' : '';
+
+			$this->start_controls_tab(
+				"_tab_positioning{$tab}",
+				[
+					'label' => '' === $tab ? esc_html__( 'Normal', 'elementor' ) : esc_html__( 'Hover', 'elementor' ),
+				]
+			);
+
+			$this->add_control(
+				"_transform_rotate_popover{$tab}",
+				[
+					'label' => esc_html__( 'Rotate', 'elementor' ),
+					'type' => Controls_Manager::POPOVER_TOGGLE,
+					'prefix_class' => $transform_prefix_class,
+					'return_value' => $transform_return_value,
+				]
+			);
+
+			$this->start_popover();
+
+			$this->add_responsive_control(
+				"_transform_rotateZ_effect{$tab}",
+				[
+					'label' => esc_html__( 'Rotate', 'elementor' ),
+					'type' => Controls_Manager::SLIDER,
+					'size_units' => [ 'deg', 'grad', 'rad', 'turn' ],
+					'default' => [
+						'unit' => 'deg',
+					],
+					'device_args' => $default_unit_values_deg,
```

Comment:
@rami-elementor In my update, I have included all active breakpoints. In the version from your team not all active breakpoints were included.